src/Domain/CampaignService.php

```diff
@@ -260,4 +258,14 @@ public function checkCampaignCanBeHandledByMatchbotDB(array $campaignData, Sales
         Assertion::null($mbDomainCampaign?->getId());
         Assertion::null($mbDomainCharity?->getId());
     }
+
+    /**
+     * Gets the amount raised for a given charity campaign, based on donations in the Matchbot DB
+     *
+     * @return Money
+     */
+    public function amountRaised(int $campaignId): Money
```

Comment:
IMHO it would be a bit nicer to take a Campaign object param instead of int, since the caller already has it. Possibly same for the repository method although that's borderline. 

That would also allow easily moving the logic to return zero for a non-persisted campaign into here.

But definitely not a blocker and I think I'd be happier with this merged quickly rather than necessarily changing that so I can fix the conflict with the caching PR and merge that too.